src/pipdeptree/_models/package.py

```diff
@@ -118,31 +114,75 @@ class DistPackage(Package):
 
     """
 
-    def __init__(self, obj: DistInfoDistribution, req: ReqPackage | None = None) -> None:
-        super().__init__(obj)
+    def __init__(self, obj: Distribution, req: ReqPackage | None = None) -> None:
+        super().__init__(obj.metadata["Name"])
+        self._obj = obj
         self.req = req
-        self._project_name = ""
 
     def requires(self) -> list[Requirement]:
-        return self._obj.requires()  # type: ignore[no-untyped-call,no-any-return]
+        req_list = []
+        req_name_list = []
+        if self._obj.requires:
+            for r in self._obj.requires:
+                req = Requirement(r)
+                is_extra_req = req.marker and contains_extra(str(req.marker))
+                if not is_extra_req and req.name not in req_name_list:
+                    req_list.append(req)
+                    req_name_list.append(req.name)
+        return req_list
 
     @property
-    def project_name(self) -> str:
-        if not self._project_name:
-            try:
-                self._project_name = metadata(self.key)["name"]
-            except (PackageNotFoundError, KeyError):
-                self._project_name = self._obj.project_name
-        return self._project_name
+    def editable(self) -> bool:
+        return bool(self.editable_project_location)
+
+    @property
+    def direct_url(self) -> DirectUrl | None:
+        direct_url_metadata_name = "direct_url.json"
+        result = None
+
+        try:
+            j_content = self._obj.read_text(direct_url_metadata_name)
+        except FileNotFoundError:  # pragma: no cover
+            return result
+        try:
+            if j_content:
+                result = DirectUrl.from_json(j_content)
+
+        except (
+            UnicodeDecodeError,
+            json.JSONDecodeError,
```

Comment:
After reading the `PathDistribution.read_text()` [code](https://github.com/python/cpython/blob/3.8/Lib/importlib/metadata.py#L488) and the method comment from `Distribution.read_text()`, they suppress exceptions including `FileNotFound`. This is true for v3.8 onwards. So it looks like we don't need to catch it.

(Nitpick) Let's change `j_content` to `json_str`.

(Nitpick) Let's also just import JSONDecodeError directly from `json`.

Reply:
> After reading the PathDistribution.read_text() [code](https://github.com/python/cpython/blob/3.8/Lib/importlib/metadata.py#L488) and the method comment from Distribution.read_text(), I see that they suppress exceptions including FileNotFound and just return None for any failures. This is true for v3.8 onwards. So it looks like we don't need to catch it.
So this code is still in `pip` because of compatibility? There is no problem that we remove this.

Reply:
Done